app/views/collection_drafts/forms/_descriptive_keywords.html.erb

```diff
@@ -34,6 +45,30 @@
                   <%= hidden_field_tag("draft[science_keywords][#{index}][variable_level_3]", science_keyword.fetch('VariableLevel3', '')) %>
                   <%= hidden_field_tag("draft[science_keywords][#{index}][detailed_variable]", science_keyword.fetch('DetailedVariable', '')) %>
                 </li>
+                <% selected_keywords_index += 1 %>
+              <% end %>
+
+              <% unless @keyword_recommendations.blank? %>
+                <% @keyword_recommendations.each_with_index do |recommended_keyword, recommendation_index| %>
+                  <% recommended_keyword_array = recommended_keyword.split('>') %>
+                  <% recommended_keyword_string = recommended_keyword_array.join(' > ') %>
+                  <% next if science_keywords.any? { |stored_science_keyword| keyword_string(stored_science_keyword) == recommended_keyword_string } %>
```

Comment:
Does the logic for matching keywords belong in the controller?

Reply:
I will consider this. Doing it before getting to the view would certainly prevent one of the edge cases we discussed in the morning (if all recommendations match previously selected keywords). I would prefer to do it in a concern but at this point I think the helper methods are scattered in a few different locations.

Reply:
moved this around, see if it is along the lines of what you meant